bluemira/magnetostatics/polyhedral_prism.py

```diff
@@ -0,0 +1,796 @@
+        """
+        Magnetic field
+        """
+        return _field_bottura(*args)
+
+    @staticmethod
+    def vector_potential(*args) -> np.ndarray:
+        """
+        Vector potential
+        """
+        return _vector_potential_bottura(*args)
+
+
+@nb.jit(nopython=True, cache=True)
+def _vector_norm_eps(r: np.ndarray) -> float:
+    """
+    Dodge singularities in omega_t and line_integral when field point
+    lies on an edge.
+
+    Introduces an error which is negligible provided the volume is small
+    the error is only introduced in a cylindrical volume of radius EPS**1/2
+    around the edge.
+
+    Parameters
+    ----------
+    r:
+        Vector coordinates (3)
+
+    Returns
+    -------
+    Vector norm
+
+    Notes
+    -----
+    \t:math:`\\lvert \\mathbf{r}\\rvert = \\sqrt{\\lvert \\mathbf{r}\\rvert^2+\\epsilon^2}`
+    """  # noqa: W505 E501
+    r_norm = np.linalg.norm(r)
+    return np.sqrt(r_norm**2 + ZERO_DIV_GUARD_EPS)
+
+
+@nb.jit(nopython=True, cache=True)
+def _omega_t(r: np.ndarray, r1: np.ndarray, r2: np.ndarray, r3: np.ndarray) -> float:
+    """
+    Solid angle seen from the calculation point subtended by the face
+    triangle normal must be pointing outwards from the face
+
+    Parameters
+    ----------
+    r:
+        Point at which the field is being calculated
+    r1:
+        First point of the triangle
+    r2:
+        Second point of the triangle
+    r3: float
+        Third point of the triangle
+
+    Returns
+    -------
+    Solid angle [rad]
+
+    Notes
+    -----
+    \t:math:`\\Omega_{T} = 2\\textrm{arctan}\\dfrac{(\\mathbf{r_{1}}-\\mathbf{r})\\cdot(\\mathbf{r_{2}}-\\mathbf{r})\\times(\\mathbf{r_{3}}-\\mathbf{r})}{D}`
+
+    with:
+
+    \t:math:`D=\\lvert \\mathbf{r_1}-\\mathbf{r}\\rvert\\lvert\\mathbf{r_2}-\\mathbf{r} \\rvert\\lvert \\mathbf{r_3} - \\mathbf{r}\\rvert + \\lvert \\mathbf{r_3} - \\mathbf{r}\\rvert (\\mathbf{r_1}-\\mathbf{r}) \\cdot (\\mathbf{r_2} - \\mathbf{r}) + \\lvert\\mathbf{r_2} - \\mathbf{r} \\rvert (\\mathbf{r_1}-\\mathbf{r})\\cdot(\\mathbf{r_3}-\\mathbf{r}) + \\lvert \\mathbf{r_1} - \\mathbf{r}\\rvert (\\mathbf{r_2} - \\mathbf{r}) \\cdot (\\mathbf{r_3} - \\mathbf{r})`
+
+    noting that the normal vector (outwards from the triangle) is:
+
+    \t:math:`\\mathbf{n_T} = \\dfrac{(\\mathbf{r_2} - \\mathbf{r_1}) \\times (\\mathbf{r_3} - \\mathbf{r_1})}{\\lvert (\\mathbf{r_2} - \\mathbf{r_1}) \\times (\\mathbf{r_3} - \\mathbf{r_1}) \\rvert}`
+    """  # noqa: W505 E501
+    r1_r = r1 - r
+    r2_r = r2 - r
+    r3_r = r3 - r
+
+    r1r = _vector_norm_eps(r1_r)
+    r2r = _vector_norm_eps(r2_r)
+    r3r = _vector_norm_eps(r3_r)
+    d = (
+        r1r * r2r * r3r
+        + r3r * np.dot(r1_r, r2_r)
+        + r2r * np.dot(r1_r, r3_r)
+        + r1r * np.dot(r2_r, r3_r)
+    )
+    a = np.dot(r1_r, np.cross(r2_r, r3_r))
+    # Not sure this is an issue...
+    # if abs(a) < ZERO_GUARD_EPS and (-ZERO_GUARD_EPS < d < 0):
+    #     return 0 # and not pi as per IEEE
```

Comment:
Nothing turned up so far. I'm of the opinion that this comment deserves to exist, because it remains true, and is a potential starting point if something weird does ever crop up